### PR TITLE
Rename lead-acid and lithium-ion

### PR DESCRIPTION
**Pull request by priyanshuone6**
# Description

Renamed lead-acid to lead_acid and lithium-ion to lithium_ion as discussed over [slack](https://pybamm.slack.com/archives/C010LMK2ZB9/p1619188113091100).
Add warnings and tests

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by priyanshuone6** [comment]
I guess I have renamed everything, @tinosulzer @rtimms could you please review it, thanks!

File: .gitignore [hunk 1]
```diff
@@ -104,6 +104,9 @@ KLU_module_deps
 # setup
 setup.log
 
+# test
+test.c
```
Comment (tinosulzer) [inline .gitignore:108]: why was this added?

**Review by tinosulzer — APPROVED** [review]
Thanks @priyanshuone6

File: .gitignore (hunk 1 above)
Comment (priyanshuone6) [reply .gitignore:108]: This file is generated every time someone runs the `run-tests.py`, so I thought to put it in the gitignore as it is only used locally and should not be pushed. Should I undo this?

Comment (tinosulzer) [reply .gitignore:108]: ok, no problem that's fine
